helpers/class.TestSession.php

```diff
@@ -719,7 +719,7 @@ public function closeTimer($identifier, $type = null)
             $source = $constraint->getSource();
             $placeId = $source->getIdentifier();
             if ($placeId === $identifier) {
-                if (($timeLimits = $source->getTimeLimits()) !== null && ($maxTime = $timeLimits->getMaxTime()) !== null) {
+                if (($maxTime = $constraint->getAdjustedMaxTime()) !== null) {
```

Comment:
Method `getAdjustedMaxTime` is not complaint with `TimeConstraint` interface from qtism  which we use. Can we use one of existing methods here? Otherwise we have to add this method to the interface as well.

Reply:
`QtiTimeConstraint` extends `TimeContstraint`. I do not see a good candidate method from either of these classes that would make sense to modify. The adjustment has to be applied to the `TimeLimts::$maxTime` value, and TimeConstraint I think is a proper place do it.
It is not uncommon in `taoQtiTest` implementation to add new public methods that make sense for TAO.